docs/modules/ROOT/pages/configuration.adoc

```diff
@@ -23,30 +29,104 @@ so the database connections are correctly configured.
 like being up to a couple of seconds out is ok. This is fine for `edge-rest` as it's major functionality
 is reading via a GET. To use a read replica `db.` prefixes  use `db-replica` prefixes to configure a read replica,
 where it is and how it should be connected to. Typically an `edge-rest` deployment will configure both of these (`db`
 and `db-replica` parameters) but the corresponding `mr` will not.
 
-== NATS
+== Async options for FeatureHub
+
+If you use a streaming version of FeatureHub, you need an async layer and this has traditionally been NATS.
+With the advent of Dacha2, the async layer is now a bigger topic. 
+
+NOTE: If you use Dacha1, you do not need to change anything, the preconfigured settings are all normal.
+
+To swap from Dacha1 (NATS only) to Dacha2 (all options), you need to set the environment variables:
+
+- `dacha1.enabled=false` - tell FeatureHub we are no longer using Dacha1
+- `dacha2.enabled=true`
+
+=== NATS
+
+The upside of NATS is simplicity and speed, it is the fastest product we test with by far. We do not use JetStream,
+but you can configure it with JetStream if you wish. The downside is typically that your Operations team has to support
+it and it is an extra cost to deploy. The Kubernetes charts for NATS are extremely good and reliable.
 
-If you are using the Streaming version of FeatureHub, then you may
-need to configure your NATS urls. If you have only once instance of a
-party-server, you can leave it as the default.  If you have deployed
-Option 2, or you have multiple servers with Option 1a, you will need to make sure  your NATS
-servers are configured correctly.
+==== Communication
+NATS communicates using a protocol whereby the clients essentially tell the servers about each other, but they
+do have to be able to communicate. Please read the https://docs.nats.io/[NATS documentation] for more information. 
+
+If you have only once instance of a
+party-server you do not need to set anything. If you have deployed NATS yourself or are running multiple Party Servers,
+you will need to set the environment variable:
 
 - `nats.urls` - a comma separated list of NATs servers.
 
-NATS works by having the clients tell the servers where each other
-are, so the NATS servers need to be routable (i.e. they must be able to talk to each other) but do not need to
-be explicitly told about each other.
+You do not need to configure anything further for NATS for Dacha1 or Dacha2. NATS supports request/reply, 
+true pub/sub and pub/queues in a single product. 
+
+=== Google's Pub/Sub
+
+For Google's PubSub, you will need to turn nats *off*. We consider Google Pub/Sub to be production ready as of 1.5.9.
+
+- `nats.enabled=false`
+
+And PubSub on:
+
+- `cloudevents.pubsub.enabled=true`
+
+You will also need to tell FeatureHub what the topics are for publishing on. There is no harm in configuring these
+across all three applications:
```

Comment:
If it is optional to configure across three applications, could we say that? Otherwise it's a bit unclear if it is mandatory or optional